api/maven-api-core/src/main/java/org/apache/maven/api/SourceRoot.java

```diff
@@ -26,13 +26,23 @@
 /**
  * A root directory of source files.
  * The sources may be Java main classes, test classes, resources or anything else identified by the scope.
+ *
+ * <h2>Default values</h2>
+ * The default implementation of all methods in this interface match the default values
+ * documented in {@code maven.mdo} (the <abbr>POM</abbr> model).
```

Comment:
I don't quite follow this, might need a bit of rewrite to clarify

Reply:
Is the following better?

> The properties in this interface are defined in the `<Source>` element of the [Maven project descriptor 4.1](https://maven.apache.org/xsd/maven-4.1.0.xsd). For each property, the default value is either empty or documented in the project descriptor. The default implementations of the methods provided in this interface are consistent with that specification.

Reply:
Can we link to something easier to read than the XSD?
Consider deleting "The default implementations of the methods provided in this interface are consistent with that specification." as that's implied by the previous sentence

Reply:
I will put a link to https://maven.apache.org/ref/current/maven-model/maven.html on the assumption that `current` will redirect to version 4.1 some day.

Reply:
What about linking to the model directly ? Everything in the xsd should be in the javadoc too ?
So maybe `{@link org.apache.maven.api.model.Model} ?

Reply:
True, I will do that.

Actually, I didn't realized that these generated classes where in public API. Since they are, shouldn't we go ahead with the use of `record` before GA, or is it too late in 4.0 release candidates for that?

Reply:
I don't think that's possible, the records cannot inherit from another record.

Reply:
However, we should change to use record-style getters (so without the `get` prefix).  But I would generate both for now and deprecate the `getXxx()` for removal in 4.1.0.